### PR TITLE
feature etb#240 lidar

### PR DESCRIPTION
**Pull request by jrachucki-n7s**

### CODE REVIEW

File: Drivers/seds/UartHwasComponent.xml [hunk 1]
```diff
@@ -1177,6 +1182,11 @@
                 <Assignment outputVariableRef="irqNumberVar">
                   <VariableRef variableRef="uartVar.mIrqNumber" />
                 </Assignment>
+                <SendCommandPrimitive shortDescription="Subscribe to uartVar interrupt" interface="InterruptSubscriptionManagement" command="SubscribeToInterrupt">
```
Comment (Lurkerpas) [inline Drivers/seds/UartHwasComponent.xml]: I have the feeling this is redundant.... Didn't we settle on making a single subscription?

File: Drivers/seds/PioHwasComponent.xml [hunk 2]
```diff
@@ -102,13 +102,13 @@
                   <VariableRef variableRef="pioVar.mPortAddress"></VariableRef>
                   <VariableRef variableRef="pioVar.mPortCodrAddress" />
                 </MathOperation>
-                <Assignment outputVariableRef="pioVar.mPortOdsrAddress">
-                  <Value value="56" />
+                <Assignment outputVariableRef="pioVar.mPortPdsrAddress">
+                  <Value value="60" />
```
Comment (Lurkerpas) [inline Drivers/seds/PioHwasComponent.xml:106]: Why this change?

Comment (jrachucki-n7s) [reply Drivers/seds/PioHwasComponent.xml:106]: This address was wrong. That's why I have changed it. It was the first SEDS where the PIO output was read, that's why this bug became apparent.

File: Drivers/seds/LidarComponent.xml [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+                  <ArgumentValue name="lidar">
+                    <VariableRef variableRef="lidarVar"></VariableRef>
+                  </ArgumentValue>
+                </SendCommandPrimitive>
+                <SendCommandPrimitive command="InitTfLunaCmd" interface="TfLuna">
+                  <LongDescription>Star UartRead</LongDescription>
+                </SendCommandPrimitive>
+              </Body>
+            </Activity>
+            <Activity name="CalibrateLidar">
```
Comment (Lurkerpas) [inline Drivers/seds/LidarComponent.xml]: Is it used?

File: Drivers/seds/LidarComponent.xml [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+                        <Assignment outputVariableRef="stepVar">
+                          <Value value="0" />
+                        </Assignment>
+                      </OnConditionFalse>
+                    </Conditional>
+                  </OnConditionTrue>
+                </Conditional>
+              </Body>
+            </Activity>
+            <Activity name="StartScanning">
```
Comment (Lurkerpas) [inline Drivers/seds/LidarComponent.xml]: Is it used?

File: Drivers/seds/UartHwasComponent.xml (hunk 1 above)
Comment (jrachucki-n7s) [reply Drivers/seds/UartHwasComponent.xml]: You are right. It should not be there. I will delete that also from SDL implementation.

File: Drivers/c/CMakeLists.txt [hunk 5]
```diff
@@ -36,6 +36,7 @@ add_subdirectory(PioHwas)
 add_subdirectory(AfecHwas)
 add_subdirectory(UartHwas)
 add_subdirectory(LidarExample)
+add_subdirectory(MP6500Example)
```
Comment (Lurkerpas) [inline Drivers/c/CMakeLists.txt]: Is this directory pushed?

File: Drivers/seds/LidarComponent.xml (hunk 3 above)
Comment (jrachucki-n7s) [reply Drivers/seds/LidarComponent.xml]: It is no longer used as there was a problem with the transition on timer. Right now, the lidar assumes that its starting position is 0 and start to move to the left, decreasing position variable on the go. After reaching the left endswitch, it sets the position variable back to 0. By doing so, it calibrates itself. 
Anyway, I can delete that or go back to using it.

File: Drivers/seds/LidarComponent.xml (hunk 4 above)
Comment (jrachucki-n7s) [reply Drivers/seds/LidarComponent.xml]: Answered above.

File: Drivers/seds/LidarComponent.xml [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+        </RequiredInterfaceSet>
+        <Implementation>
+          <VariableSet>
+            <Variable name="lidarVar" type="com/n7space/Lidar/Lidar"></Variable>
+            <Variable name="configVar" type="com/n7space/Lidar/LidarConfig" />
+            <Variable name="dataVar" type="com/n7space/Lidar/LidarData" />
+            <Variable name="tfLunaDataVar" type="com/n7space/TfLuna/TfLuna_Data" />
+            <Variable name="tfLunaErrorVar" type="com/n7space/TfLuna/TfLuna_DataFrameErr" />
+            <Variable name="errorVar" type="com/n7space/Lidar/LidarError" />
+            <Variable name="calibratedVar" type="com/n7space/Hwas/Byte" />
```
Comment (Lurkerpas) [inline Drivers/seds/LidarComponent.xml]: What is the purpose of this variable?

File: Drivers/c/CMakeLists.txt (hunk 5 above)
Comment (jrachucki-n7s) [reply Drivers/c/CMakeLists.txt]: It is not. I can delete this line or push the directory. I would prefer to delete it as this functionality works fine in the SEDS implementation ,and it is quite tivial. It just turnes the lidar left and right, changing a direction every second or so.

File: Drivers/seds/LidarComponent.xml (hunk 6 above)
Comment (jrachucki-n7s) [reply Drivers/seds/LidarComponent.xml]: Not used anymore, as I have described above.

File: Drivers/seds/Lidar.xml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+          <Enumeration label="LidarDirection_Left" value="0"></Enumeration>
+          <Enumeration label="LidarDirection_Right" value="1"></Enumeration>
+        </EnumerationList>
+      </EnumeratedDataType>
+    </DataTypeSet>
+    <DeclaredInterfaceSet>
+      <Interface level="functional" name="LidarInterfaceType">
+        <CommandSet>
+          <Command name="InitLidarCmd" mode="sync">
+            <LongDescription>Initializes UART for communication with LIDAR, PIOs of MP6500 and for end switches</LongDescription>
```
Comment (Lurkerpas) [inline Drivers/seds/Lidar.xml:51]: Please add enable/disable functionality

File: Drivers/c/CMakeLists.txt (hunk 5 above)
Comment (jrachucki-n7s) [reply Drivers/c/CMakeLists.txt]: Fixed

File: Drivers/seds/LidarComponent.xml (hunk 3 above)
Comment (jrachucki-n7s) [reply Drivers/seds/LidarComponent.xml]: fixed

File: Drivers/seds/LidarComponent.xml (hunk 4 above)
Comment (jrachucki-n7s) [reply Drivers/seds/LidarComponent.xml]: fixed

File: Drivers/seds/LidarComponent.xml (hunk 6 above)
Comment (jrachucki-n7s) [reply Drivers/seds/LidarComponent.xml]: fixed

File: Drivers/seds/Lidar.xml (hunk 7 above)
Comment (jrachucki-n7s) [reply Drivers/seds/Lidar.xml:51]: Added

**Review by Lurkerpas — APPROVED** [review]
